src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+    } else if (this->unk_332 != 0) {
+        this->unk_332--;
+    }
+    if ((globalCtx->envCtx.unk_D8 > 0.0f) && (this->unk_336 != 0)) {
+        globalCtx->envCtx.unk_E9 = 1;
+        globalCtx->envCtx.unk_EA[0] = 0xFF;
+        globalCtx->envCtx.unk_EA[1] = 0xFF;
+        globalCtx->envCtx.unk_EA[2] = 0xFF;
+        globalCtx->envCtx.unk_EA[3] = (s16)(globalCtx->envCtx.unk_D8 * 200.0f);
+    } else {
+        globalCtx->envCtx.unk_E9 = 0;
+    }
+    globalCtx->envCtx.unk_BF = 0;
+    globalCtx->envCtx.unk_DC = 2;
+
+    switch (this->unk_339) {
+        case 0:
+            Math_ApproachZeroF(&globalCtx->envCtx.unk_D8, 1.0f, 0.1f);
+            break;
+        case 3:
+            globalCtx->envCtx.unk_BE = 3;
+            globalCtx->envCtx.unk_BD = 4;
+            Math_ApproachF(&globalCtx->envCtx.unk_D8, 1.0f, 1.0f, 0.0125f);
+            break;
+        case 4:
+            globalCtx->envCtx.unk_BE = 5;
+            globalCtx->envCtx.unk_BD = 6;
+            Math_ApproachF(&globalCtx->envCtx.unk_D8, 1.0f, 1.0f, 0.0125f);
+            break;
+        case 5:
+            globalCtx->envCtx.unk_BE = 6;
+            globalCtx->envCtx.unk_BD = 7;
+            Math_ApproachF(&this->unk_33C, 0.69f, 1.0f, 0.05f);
+            globalCtx->envCtx.unk_D8 =
+                (Math_SinS(globalCtx->gameplayFrames * 0x5000) * 0.15f) + (0.15f + this->unk_33C);
+            break;
+        case 55:
+            globalCtx->envCtx.unk_BE = 2;
+            globalCtx->envCtx.unk_BD = 0;
+            Math_ApproachZeroF(&globalCtx->envCtx.unk_D8, 1.0f, 0.05f);
+            break;
+        case 6:
+            globalCtx->envCtx.unk_BE = 2;
+            globalCtx->envCtx.unk_BD = 8;
+            Math_ApproachF(&this->unk_33C, 0.69f, 1.0f, 0.05f);
+            globalCtx->envCtx.unk_D8 =
+                (Math_SinS(globalCtx->gameplayFrames * 0x7000) * 0.15f) + (0.15f + this->unk_33C);
+            break;
+        case 7:
+            globalCtx->envCtx.unk_BE = 0;
+            globalCtx->envCtx.unk_BD = 8;
+            Math_ApproachZeroF(&globalCtx->envCtx.unk_D8, 1.0f, 0.02f);
+            break;
+        case 20:
+            globalCtx->envCtx.unk_BE = 0;
+            globalCtx->envCtx.unk_BD = 9;
+            break;
+        case 21:
+            globalCtx->envCtx.unk_BE = 0xA;
+            globalCtx->envCtx.unk_BD = 9;
+            break;
+        case 22:
+            globalCtx->envCtx.unk_BE = 0xA;
+            globalCtx->envCtx.unk_BD = 0xB;
+            break;
+        case 23:
+            globalCtx->envCtx.unk_BE = 9;
+            globalCtx->envCtx.unk_BD = 0xB;
+            break;
+        case 24:
+            globalCtx->envCtx.unk_BE = 0;
+            globalCtx->envCtx.unk_BD = 0xC;
+            break;
+        case -1:
+            break;
+    }
+
+    if (this->unk_339 >= 0) {
+        this->unk_339 = 0;
+    }
+    if (D_80906D78 != 0) {
+        D_80906D78 = 0;
+
+        for (i2 = 0; i2 < 100; i2++) {
```

Comment:
```suggestion
        for (i2 = 0; i2 < ARRAY_COUNT(sParticles); i2++) {
```
Not sure about this one, kind of makes sense since this is one of the only two uses of sParticles (through `func_808FD27C` here) but also maybe you would want less particles than possible here, and having a number larger than 100 isn't an issue, just won't spawn the former effects

Reply:
I'll hold off on this change since you're not sure, but I should definitely make this change here as well since it's directly iterating over the particles? https://github.com/zeldaret/oot/blob/cb7b9fdcfaa5666651fba4be15576d14ec6ad2e9/src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c#L3276-L3283

Reply:
Yeah that for loop you linked is definitely iterating over the whole array. I'm hesitating about the one I commented on because the code would also work with any value below 100 while in most other places it needs to be exactly 100 = ARRAY_COUNT

oh well just change both I guess

Reply:
Done, changed both.